src/App/pages/HomePage/components/CategoryTabs/CategoryTabs.tsx

```diff
@@ -1,40 +1,46 @@
-import classNames from 'classnames';
-import { useEffect } from 'react';
-import { useSearchParams } from 'react-router-dom';
+import useEmblaCarousel from 'embla-carousel-react';
+import { observer } from 'mobx-react-lite';
 import Button from 'components/Button';
 import Icon from 'components/Icon';
-import { categories } from 'configs/consts';
+import { NAMED_CATEGORIES } from 'config/consts';
+import { useMatchMedia } from 'hooks';
+import { useSearchQuery } from 'hooks/useSearchQuery';
+import rootStore from 'store/RootStore';
+import { Meta } from 'store/types';
+import Tab from './components/Tab';
 import styles from './CategoryTabs.module.scss';
 
 const CategoryTabs = () => {
-  const [searchParams, setSearchParams] = useSearchParams();
+  const { isMobile } = useMatchMedia();
 
-  const currentCategory = searchParams.get('category') || 'jackets';
+  const { setQueryParam } = useSearchQuery();
 
-  useEffect(() => {
-    if (!searchParams.get('category')) {
-      setSearchParams({ category: 'jackets' });
-    }
-  }, [searchParams, setSearchParams]);
+  const { searchQueryStore, productsStore } = rootStore;
 
-  const cn = classNames(styles['CategoryTabs']);
+  const [emblaRef] = useEmblaCarousel({ dragFree: true });
+
+  const Tabs = NAMED_CATEGORIES.map((category) => (
+    <Tab
+      key={category.category}
+      category={category}
+      isActive={searchQueryStore.getParam('category') === category.category}
+      disabled={productsStore.meta === Meta.loading || productsStore.meta === Meta.initial}
+      onClick={() => setQueryParam('category', category.category)}
```

Comment:
тогда тут можно будет не инлайнить коллбэк а вынести его в `useCallback`
```ts
const handleTabClick = React.useCallback((category: TNamedCategory) => setQueryParam('category', category.category), [setQueryParam])
```
```suggestion
      onClick={handleTabClick}
```

Reply:
Исправил.